src-built-in/components/UIAPI/reducers/toolbar.ts

```diff
@@ -0,0 +1,27 @@
+// import { loop, Cmd } from 'redux-loop';
+
+// import { ActionTypes } from '../types';
+// import { loadPinnedItems } from '../effects/toolbar';
+
+// const initialState = {
+//     pins: {
+//         workspace: {
+
+//         },
+//         component: {
+
+//         }
+//     }
+// };
+
+// export const toolbar = (state = initialState, action) => {
+//     const { type, payload } = action;
+//     switch (type) {
```

Comment:
The design should include all of the actions necessary for the entire favorites subsystem. Try not to think about one component at a time. Think holistically.

INITIALIZE_FAVORITES
ADD_FAVORITE
REMOVE_FAVORITE
REORDER_FAVORITE
etc

Refer to the requirements document to determine what your actions need to be.

Reply:
Got it!